src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -447,13 +611,302 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector64_get_Zero:
+        case NI_Vector128_get_Zero:
+        {
+            assert(sig->numArgs == 0);
+            retNode = gtNewSimdZeroNode(retType, simdBaseJitType, simdSize, /* isSimdAsHWIntrinsic */ false);
+            break;
+        }
+
+        case NI_Vector64_GetElement:
+        case NI_Vector128_GetElement:
+        {
+            assert(!sig->hasThis());
+            assert(numArgs == 2);
+
+            op2 = impPopStack().val;
+            op1 = impSIMDPopStack(getSIMDTypeForSize(simdSize));
+
+            const bool isSimdAsHWIntrinsic = true;
+            retNode = gtNewSimdGetElementNode(retType, op1, op2, simdBaseJitType, simdSize, isSimdAsHWIntrinsic);
+            break;
+        }
+
+        case NI_Vector128_GetUpper:
```

Comment:
I don't think this is the most optimal way of implementing `Vector128.GetUpper()`...

Perhaps, something that does `DUP Vd, Vn.D[1]` should be used instead.
This would correspond to `vector.AsUInt64().GetElement(1).As<T>()`

Reply:
Do you mind if I fix this in a follow up PR (just so we can go ahead and get this one resolved and avoid further downstream conflicts)?

Also noting, this might be a case where we want to recognize the `ExtractVector128` pattern and replace it with `DUP` if we know its always going to be more efficient...

Reply:
Sure, I meant to write that this work can wait when posted the message (but forgot).